### ISSUE TITLE
org.apache.avro.SchemaParseException when generic type has arguments with same name

### ISSUE THREAD

**Issue by IvanStepanov** [sksamuel/avro4s#356]
If i define the following types

```
object First {
  case class TypeArg(a: String)
}

object Second {
  @AvroName("TypeArg2") // doesn't change anything
  case class TypeArg(a: String)
}

case class Generic[T](value: T)
case class MyClass(a: Generic[First.TypeArg], b: Generic[Second.TypeArg])
```

and then try to generate schema string
```
val schema = AvroSchema[MyClass]
println(schema.toString(true))
```

then i will get ```org.apache.avro.SchemaParseException: Can't redefine: mynamespace.TestApp.Generic__TypeArg```

It will properly work with generic type parameters with different names (for ex ```String``` and ```Int```), but will stuck  on types with same name (```TypeArg``` from different objects in my case).

Probably some kind of full name should be used for resulting avro type name, or at least ```@AvroName``` annotation should redefine type arg name in generic name?

**Comment by bobbyrauchenberg** [comment]
you could do
```
object First {
  case class TypeArg(a: String)
}

object Second {
  case class TypeArg(a: String)
}

case class Generic[T](value: T)
case class MyClass(a: Generic[First.TypeArg], @AvroNamespace("second") b: Generic[Second.TypeArg])
```

**Comment by sksamuel** [comment]
Released 3.0.0 which supports @AvroName on a type arg.
